dialect/dialect.go

```diff
@@ -68,6 +68,12 @@ type DebugDriver struct {
 	log    func(...interface{}) // log function. defaults to log.Println.
 }
 
+// DebugDriverWithContext is a driver that logs all driver operations with context.
+type DebugDriverWithContext struct {
+	Driver                                       // underlying driver.
+	logCtx func(context.Context, ...interface{}) // log function.
+}
+
```

Comment:
Maybe we can merge these 2 drivers into one as follows:

```go
// DebugDriver is a driver that logs all driver operations.
type DebugDriver struct {
	Driver                                       // underlying driver.
	log    func(context.Context, ...interface{}) // log function. defaults to log.Println.
}

// Debug gets a driver and an optional logging function, and returns
// a new debugged-driver that prints all outgoing operations.
func Debug(d Driver, logger ...func(...interface{})) Driver {
	logf := log.Println
	if len(logger) == 1 {
		logf = logger[0]
	}
	return &DebugDriver{d, func(_ context.Context, v ...interface{}) { logf(v...) }}
}

// DebugWithContext gets a driver and an logging function, and returns
// a new debugged-driver that prints all outgoing operations with context.
func DebugWithContext(d Driver, logger func(context.Context, ...interface{})) Driver {
	return &DebugDriver{d, logger}
}
```

Reply:
I was thinking the same. 
I just didn't know if it was ok to change struct signature.
I'll try to fix this.